File: NotificationMS/src/main/java/com/alumniconnect/notification/dto/NotificationDto.java
package com.alumniconnect.notification.dto;

import com.alumniconnect.notification.enums.NotificationType;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class NotificationDto {

    @NotNull(message = "User Id is required")
    private Long userId;

    @NotBlank(message = "Title is required")
    private String title;

    @NotBlank(message = "Message is required")
    private String message;

    @NotNull(message = "Notification Type is required")
    private NotificationType type;

    private Boolean isRead = false;

    private String createdAt;
}

File: NotificationMS/DTOs/NotificationDto.cs
using System.ComponentModel.DataAnnotations;
using NotificationMS.Enums;

namespace NotificationMS.DTOs
{
    public class NotificationDto
    {
        [Required(ErrorMessage = "User Id is required")]
        public long? UserId { get; set; }

        [Required(ErrorMessage = "Title is required")]
        public string Title { get; set; } = string.Empty;

        [Required(ErrorMessage = "Message is required")]
        public string Message { get; set; } = string.Empty;

        [Required(ErrorMessage = "Notification Type is required")]
        public NotificationType? Type { get; set; }

        public bool IsRead { get; set; } = false;

        public string? CreatedAt { get; set; }
    }
}
